### PR TITLE
🌱 Backport re-pivoting e2e test

### PR DESCRIPTION
**Pull request by Rozzii**
The main goal of this commit to implement the re-pivoting e2e
test.

This commit also renames the e2e test functions to be aligned
with GO naming conventions.

### CODE REVIEW

File: test/e2e/e2e_test.go
Comment (mboukhalfa) [inline test/e2e/e2e_test.go:84]: Can we ignore remediation

Comment (Rozzii) [reply test/e2e/e2e_test.go:84]: Is it something that we don't want in release-0.5?

Comment (mboukhalfa) [reply test/e2e/e2e_test.go:84]: no just to test cluster delete quickly but if you think that tests will pass then should be kept in the test

**Comment by furkatgofurov7** [comment]
I will debug some more things once again within the running VM. 

/test-v1a5-e2e

**Review by namnx228 — APPROVED** [review]
/lgtm

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:268]:
```suggestion
	Expect(err).To(BeNil(), "Failed to delete Ironic from the target cluster")
```

**Comment by furkatgofurov7** [comment]
> This commit also adds the process of setting the nodeReuse
> variable of metal3MachineTemplates to "false" in order to
> fix target cluster cleanup issue that is present the value is "true"
> instead of "false".

You have to remove these notes from the PR description?

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [reply test/e2e/pivoting_test.go:268]: This will be covered as part of #390 when I will backport it to release-0.5, so I am fine merging this as is.

File: test/e2e/pivoting_test.go
Comment (kashifest) [inline test/e2e/pivoting_test.go:319]: what if we fail in any of these three phases in else block? Wouldn't if be good to have some phase specific error log here or the phases take care of this separately? By phase I mean, `configureIronicConfigmap, installIronicBMO and configureIronicConfigmap`

**Review by kashifest — COMMENTED** [review]
/lgtm

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go:319]: Yes, the phases take care of error logs separately, the phases contain "Expect" function calls that are wrappers for the assertions and if any of them fails then we get a custom error message, the expected and the received value comparison and the full stack trace as output so we know exactly what failed.

**Comment by Rozzii** [comment]
/ test-v1a5-e2e

**Review by furkatgofurov7 — APPROVED** [review]
/approve

**Comment by furkatgofurov7** [comment]
/hold
Until #354 is merged, this is not a backport, let's merge main PR and remove hold and consider it as backport. Meanwhile let's try to get all tests to green.

**Review by namnx228 — COMMENTED** [review]
/lgtm

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:381]:
```suggestion
		machines := &capi.MachineList{}
```

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:295]: Let's add a log for beginning of re-pivoting tests here as well?

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [reply test/e2e/pivoting_test.go:269]: *target

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:340]:
```suggestion
	By("Check that the re-pivoted cluster is up and running")
```

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:347]:
```suggestion
	By("Check that the control plane of the re-pivoted cluster is up and running")
```

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:329]:
```suggestion
	By("Move back to bootstrap cluster")
```

File: test/e2e/pivoting_test.go
Comment (furkatgofurov7) [inline test/e2e/pivoting_test.go:296]:
```suggestion
	By("Remove Ironic deployment from target cluster")
```

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Comment by furkatgofurov7** [comment]
It would be easier just to use cherry pick bot from the main branch IMO:)

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go]: fixed

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go:295]: fixed

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go]: fixed

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go]: fixed

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go]: fixed

File: test/e2e/pivoting_test.go
Comment (Rozzii) [reply test/e2e/pivoting_test.go]: fixed

**Comment by furkatgofurov7** [comment]
For some reason untainting of CP nodes were not successful and this is the first time I am seeing the similar problem in e2e. Re-triggering to see if it happens again.

/test-v1a5-centos-e2e

**Comment by furkatgofurov7** [comment]
Unrelated to e2e flake while repo fetching.

/test-v1a5-centos-e2e

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**
